contracts/Removal.sol

```diff
@@ -46,6 +46,18 @@ contract Removal is
     private _addressToOwnedTokenIds;
   uint256 private _currentMarketBalance;
 
+  /**
+   * @notice Emitted on updating `_removalIdToRemovalData` with a new removal.
+   * @param removalId The id of the removal that was released.
+   * @param releasedFrom The address the removal was released from.
+   * @param amount The amount that was released.
+   */
+  event RemovalReleased(
+    uint256 indexed removalId,
+    address indexed releasedFrom,
```

Comment:
naming feels redundant given the name of the event, perhaps change to `from`, `account` `owner` `supplier` or something like that